Jenkinsfile

```diff
@@ -470,355 +453,395 @@ def testGo(prefix, packagesToTest) {
-        ],
-      ],
-      test_windows_go_: [
-        '*': [],
-        'github.com/keybase/client/go/systests': [
-          disable: true,
-        ],
-        'github.com/keybase/client/go/chat': [
-          disable: true,
-        ],
+def testGoTestSuite(prefix, packagesToTest) {
+  def dirs = getTestDirsNix()
+  def goversion = sh(returnStdout: true, script: "go version").trim()
+  println "Testing Go code on commit ${env.COMMIT_HASH} with ${goversion}. Merging to branch ${env.CHANGE_TARGET}."
+
+  // Make sure we don't accidentally pull in the testing package.
+  sh '! go list -f \'{{ join .Deps "\\n" }}\' github.com/keybase/client/go/keybase | grep testing'
+
+  println "Building citogo"
+  sh '(cd citogo && go install)'
+
+  def packageTestList = packagesToTest.keySet()
+  println "Go packages to test:\n${packageTestList.join('\n')}"
+
+  def tests = [:]
+  def specialTests = [:]
+  def specialTestFilter = [
+      'chat', 'engine', 'teams', 'chat_storage', 'systests', 'kbfs_libdokan',
+      'kbfs_test_race', 'stellar_stellarsvc', 'tlfupgrade', 'service',
+      'saltpackkeys', 'kbfs_libkbfs', 'kbfs_test', 'identify3', 'git',
+      'ephemeral'
+  ]
+  def testSpecMap = [
+    test_macos_go_: [
+      'github.com/keybase/client/go/kbfs/test': [
+        name: 'kbfs_test_fuse',
+        flags: '-tags fuse',
+        timeout: '15m',
       ],
-    ]
-    def getOverallTimeout = { testSpec ->
-      def timeoutMatches = (testSpec.timeout =~ /(\d+)([ms])/)
-      return [
-        time: 1 + (timeoutMatches[0][1] as Integer),
-        unit: timeoutMatches[0][2] == 's' ? 'SECONDS' : 'MINUTES',
-      ]
-    }
-    def defaultPackageTestSpec = { pkg ->
-      def dirPath = pkg.replaceAll('github.com/keybase/client/go/', '')
-      def testName = dirPath.replaceAll('/', '_')
-      return [
-        name: testName,
+      'github.com/keybase/client/go/kbfs/libfuse': [
+        timeout: '3m',
+      ],
+      'github.com/keybase/client/go/libkb': [
+        timeout: '5m',
+      ],
+      'github.com/keybase/client/go/install': [
+        timeout: '30s',
+      ],
+      'github.com/keybase/client/go/launchd': [
+        timeout: '30s',
+      ],
+    ],
+    test_linux_go_: [
+      '*': [],
+      'github.com/keybase/client/go/kbfs/test': [
+        name: 'kbfs_test_fuse',
+        flags: '-tags fuse',
+        timeout: '15m',
+      ],
+      'github.com/keybase/client/go/kbfs/data': [
+        flags: '-race',
+        timeout: '30s',
+      ],
+      'github.com/keybase/client/go/kbfs/libfuse': [
         flags: '',
-        timeout: '30m',
-        dirPath: dirPath,
-      ]
-    }
-    def getPackageTestSpec = { pkg ->
-      if (testSpecMap[prefix].containsKey(pkg)) {
-        if (testSpecMap[prefix][pkg]) {
-          def testSpec = testSpecMap[prefix][pkg]
-          if (testSpec['disable']) {
-            return false
-          }
-          return defaultPackageTestSpec(pkg) + testSpec
+        timeout: '15m',
```

Comment:
i don't think this is the issue, i think the libfuse tests are indefinitely deadlocked.

Reply:
(when then fail in this way)

Reply:
If so, then I don't see how reverting to the old approach would be helpful.

Reply:
Got it, I thought it was due to my PR. I'll revert